test/run-spec-wasm2c.py

```diff
@@ -101,11 +101,21 @@ def MangleTypes(types):
 def MangleName(s):
     def Mangle(match):
         s = match.group(0)
-        return b'Z%02X' % s[0]
+        return b'0x%02X' % s[0]
 
-    # NOTE(binji): Z is not allowed.
-    pattern = b'([^_a-zA-Y0-9])'
-    return 'Z_' + re.sub(pattern, Mangle, s.encode('utf-8')).decode('utf-8')
+    # escape underscores at beginning and end
+    pattern = b'((^_)|(_$))'
+    s = re.sub(pattern, Mangle, s.encode('utf-8')).decode('utf-8')
+
+    # NOTE(keithw): forced escapes for '0x[hexdigit]' not implemented here
+    pattern = b'([^_a-zA-Z0-9])'
+    return re.sub(pattern, Mangle, s.encode('utf-8')).decode('utf-8')
+
+
+def MangleModuleName(s):
+    # double underscores
+    return MangleName(re.sub(b'(_)', b'__',
+                             s.encode('utf-8')).decode('utf-8'))
```

Comment:
This roundtrip through encode and decode looks odd.

Can we not just stay in decoded the whole time by doing `re.sub('(_)', '__', s)`?

Reply:
Yes, done in 3e5e00f9438390b4e0a368f397323573f9c24eb6